src/runtime/pythonexception.cs

```diff
@@ -261,10 +261,11 @@ internal static void ThrowIfIsNull(BorrowedReference reference)
             }
         }
 
-        public static void ThrowIfIsNotZero(int value)
+        public static void ThrowIfIsNotZero(int value, Action onBeforeThrow = null)
         {
             if (value != 0)
             {
+                onBeforeThrow?.Invoke();
```

Comment:
That's overengineered.

If you have
```csharp
if (value != 0)
{
  doStuff();
  throw;
}
```
pattern, it is better to spell it out (IMHO), than pass a lambda to a call. Besides lambdas have relatively significant cost to construct, whilst being totally unnecessary in 99% of cases when the call does not fail.

Reply:
@lostmsu Fair point, I thought it wouldn't hurt to "propose" it.  Since the call usually doesn't fail I figured the branch predictor would optimize it well enough.